### PR TITLE
Wal e recovery

### PR DESCRIPTION
**Pull request by jharveysmith**
Add LWRP for wal-e envdir.
Add seperate recovery and archive recipes for wal-e.

@mbautin 
@vnivargi

### CODE REVIEW

**Comment by jharveysmith** [comment]
We should probably merge this.

File: providers/wal-e_envdir.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  s3path = case
+    when new_resource.s3path
+      new_resource.s3path
+    when (node['postgresql']['wal_e']['s3_path'] || '') =~ /^[sS]3:/
+      node['postgresql']['wal_e']['s3_path']
+    when node['postgresql']['wal_e']['s3_path']
+      "s3://" + node['postgresql']['wal_e']['s3_path']
+    else
+      "s3://#{node['postgresql']['wal_e']['s3_bucket']}/#{node['postgresql']['wal_e']['bkp_folder']}"
+    end
```
Comment (mbautin) [inline providers/wal-e_envdir.rb]: Unindent `end` by two spaces?

File: providers/wal-e_envdir.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  directory mypath do
+    user    myuser
+    group   mygroup
+    mode    "0550"
+  end
+
+  vars.each do |key, value|
+    file ::File.join(mypath,key) do
```
Comment (mbautin) [inline providers/wal-e_envdir.rb]: Nit: add a space after comma.

File: recipes/wal-e_recovery.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  # Destroy the existing data_directory and restore from our basebackup.
+  bash "restory_catalog-service_basebackup" do
+    cwd "/"
+    user  myuser
+    group mygroup
+    command <<-EOH
+      rm -rf #{node['postgresql']['config']['data_directory']}/*
+      envdir #{env_dir} /usr/local/bin/wal-e backup-fetch #{node['postgresql']['config']['data_directory']} #{node['postgresql']['recovery']['base_backup_target']}
+    EOH
```
Comment (mbautin) [inline recipes/wal-e_recovery.rb]: Fix long line. You can use `\` to break long lines.

**Comment by mbautin** [comment]
LGTM. A few minor comments inline.

**Comment by jharveysmith** [comment]
Good to merge?

**Pull request merged**